### PR TITLE
servicemonitor add batch delete

### PR DESCRIPTION
**Pull request by ifooth**

### CODE REVIEW

File: bcs-app/backend/apps/metric/views/servicemonitor.py [hunk 1]
```diff
@@ -331,6 +331,34 @@ def update(self, request, project_id, cluster_id, namespace, name):
         self._activity_log(project_id, request.user.username, name, message, True)
         return Response(result)
 
+    def bacth_delete(self, request, project_id, cluster_id):
+        """批量删除
+        """
+        slz = serializers.ServiceMonitorBatchDeleteSLZ(data=request.data)
+        slz.is_valid(raise_exception=True)
+        data = slz.validated_data["servicemonitors"]
```
Comment (jamesgetx) [inline bcs-app/backend/apps/metric/views/servicemonitor.py:339]: 可以命名具体化成svc_monitors

File: bcs-app/backend/apps/metric/views/servicemonitor.py [hunk 2]
```diff
@@ -331,6 +331,34 @@ def update(self, request, project_id, cluster_id, namespace, name):
...
+
+        ns_list = [(cluster_id, i["namespace"]) for i in data]
+        self._validate_namespace_use_perm(request, project_id, ns_list)
+
+        client = self._get_client(request, project_id, cluster_id)
+        succeed = []
+        for servicemonitor in data:
```
Comment (jamesgetx) [inline bcs-app/backend/apps/metric/views/servicemonitor.py:346]: `for monitor in svc_monitors:`

File: bcs-app/backend/apps/metric/views/servicemonitor.py [hunk 3]
```diff
@@ -331,6 +331,34 @@ def update(self, request, project_id, cluster_id, namespace, name):
...
+        """
+        slz = serializers.ServiceMonitorBatchDeleteSLZ(data=request.data)
+        slz.is_valid(raise_exception=True)
+        data = slz.validated_data["servicemonitors"]
+
+        ns_list = [(cluster_id, i["namespace"]) for i in data]
+        self._validate_namespace_use_perm(request, project_id, ns_list)
+
+        client = self._get_client(request, project_id, cluster_id)
+        succeed = []
```
Comment (jamesgetx) [inline bcs-app/backend/apps/metric/views/servicemonitor.py:345]: 来个复数？

**Review by jamesgetx — APPROVED** [review]

File: bcs-app/backend/apps/metric/views/servicemonitor.py (hunk 1 above)
Comment (ifooth) [reply bcs-app/backend/apps/metric/views/servicemonitor.py]: done

File: bcs-app/backend/apps/metric/views/servicemonitor.py (hunk 3 above)
Comment (ifooth) [reply bcs-app/backend/apps/metric/views/servicemonitor.py]: done

File: bcs-app/backend/apps/metric/views/servicemonitor.py (hunk 2 above)
Comment (ifooth) [reply bcs-app/backend/apps/metric/views/servicemonitor.py]: done

**Review by jamesgetx — APPROVED** [review]

**Pull request merged**